pyuvdata/tests/test_uvdata.py

```diff
@@ -3909,3 +3914,1014 @@ def test_lsts_from_time_with_only_unique():
+
+    # reorder to make sure we get the right value later
+    uv_object.reorder_blts(order="baseline")
+
+    # save some values for later
+    init_data_size = uv_object.data_array.size
+    init_wf_01 = uv_object.get_data(0, 1)
+    init_wf_02 = uv_object.get_data(0, 2)
+    # check that there are no flags
+    assert np.nonzero(uv_object.flag_array)[0].size == 0
+    init_ns_01 = uv_object.get_nsamples(0, 1)
+    init_ns_02 = uv_object.get_nsamples(0, 2)
+
+    # change the target integration time
+    max_integration_time = np.amin(uv_object.integration_time)
+    uv_object.upsample_in_time(max_integration_time, blt_order="baseline")
+
+    assert np.allclose(uv_object.integration_time, max_integration_time)
+    # output data should be the same
+    out_wf_01 = uv_object.get_data(0, 1)
+    out_wf_02 = uv_object.get_data(0, 2)
+    assert np.all(init_wf_01 == out_wf_01)
+    assert np.isclose(init_wf_02[0, 0, 0], out_wf_02[0, 0, 0])
+    assert init_wf_02.size * 2 == out_wf_02.size
+
+    # this should be true because there are no flags
+    out_ns_01 = uv_object.get_nsamples(0, 1)
+    out_ns_02 = uv_object.get_nsamples(0, 2)
+    assert np.allclose(out_ns_01, init_ns_01)
+    assert np.isclose(init_ns_02[0, 0, 0], out_ns_02[0, 0, 0])
+
+    return
+
+
+@uvtest.skipIf_no_h5py
+@pytest.mark.filterwarnings("ignore:The xyz array in ENU_from_ECEF")
+@pytest.mark.filterwarnings("ignore:The enu array in ECEF_from_ENU")
+def test_upsample_in_time_drift():
+    """Test the upsample_in_time method on drift mode data"""
+    uv_object = UVData()
+    testfile = os.path.join(DATA_PATH, 'zen.2458661.23480.HH.uvh5')
+    uv_object.read(testfile)
+
+    # reorder to make sure we get the right value later
+    uv_object.reorder_blts(order="baseline")
+    uv_object_copy = uv_object.copy()
+
+    # save some values for later
+    init_data_size = uv_object.data_array.size
+    init_wf = uv_object.get_data(0, 1)
+    # check that there are no flags
+    assert np.nonzero(uv_object.flag_array)[0].size == 0
+    init_ns = uv_object.get_nsamples(0, 1)
+
+    # change the target integration time
+    max_integration_time = np.amin(uv_object.integration_time) / 2.0
+    uv_object.upsample_in_time(max_integration_time, blt_order="baseline", allow_drift=True)
+
+    assert np.allclose(uv_object.integration_time, max_integration_time)
+    # we should double the size of the data arrays
+    assert uv_object.data_array.size == 2 * init_data_size
+    # output data should be the same
+    out_wf = uv_object.get_data(0, 1)
+    # we need a "large" tolerance given the "large" data
+    new_tol = 1e-2 * np.amax(np.abs(uv_object.data_array))
+    assert np.isclose(init_wf[0, 0, 0], out_wf[0, 0, 0], atol=new_tol)
+
+    # this should be true because there are no flags
+    out_ns = uv_object.get_nsamples(0, 1)
+    assert np.isclose(init_ns[0, 0, 0], out_ns[0, 0, 0])
+
+    # try again with allow_drift=False
```

Comment:
might even break up the with and without `allow_drift` case

Reply:
Done.